### PR TITLE
Build empty javadoc and sources jars for azure-spring-cloud-starter-appconfiguration-config

### PR DESCRIPTION
**Pull request by JimSuplizio**
Add the build rules to produce empty javadoc and sources jars for azure-spring-cloud-starter-appconfiguration-config.

### CODE REVIEW

**Review by alzimmermsft — APPROVED** [review]

**Review by srnagar — APPROVED** [review]

File: sdk/appconfiguration/azure-spring-cloud-starter-appconfiguration-config/pom.xml [hunk 1]
```diff
@@ -26,4 +26,95 @@
...
+            <goals>
+              <goal>jar</goal>
+            </goals>
+            <configuration>
+              <classifier>sources</classifier>
+              <classesDirectory>${project.basedir}/sourceTemp</classesDirectory>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
```
Comment (alzimmermsft) [inline sdk/appconfiguration/azure-spring-cloud-starter-appconfiguration-config/pom.xml:84]: I think this was supposed to be executions for maven-jar-plugin?

Comment (JimSuplizio) [reply sdk/appconfiguration/azure-spring-cloud-starter-appconfiguration-config/pom.xml]: Yeah, I've pushed a change to fix this. Yay copy and paste.

**Comment by JimSuplizio** [comment]
Because this is a POM only change, I had to kick off the pipeline [manually](https://dev.azure.com/azure-sdk/public/_build/results?buildId=1594186&view=logs&j=9d510ff3-53a8-5192-bd2d-466ce830bc13&t=01de8c6b-9f4b-5017-894e-951524cf562e&l=9) against my pull. This change fixes the build failure due to missing javadoc and sources.

File: sdk/appconfiguration/azure-spring-cloud-starter-appconfiguration-config/pom.xml [hunk 2]
```diff
@@ -26,4 +26,79 @@
...
+            <phase>prepare-package</phase>
+            <configuration>
+              <target>
+                <echo>Deleting existing ${project.basedir}/javadocTemp and
+                  ${project.basedir}/sourceTemp</echo>
+                <delete includeEmptyDirs="true" quiet="true">
+                  <fileset dir="${project.basedir}/javadocTemp"/>
+                  <fileset dir="${project.basedir}/sourceTemp"/>
+                </delete>
+                <echo>Copying ${project.basedir}/../README.md to
```
Comment (chenrujun) [inline sdk/appconfiguration/azure-spring-cloud-starter-appconfiguration-config/pom.xml:85]: `${project.basedir}/../README.md` -> `${project.basedir}/README.md`.

File: sdk/appconfiguration/azure-spring-cloud-starter-appconfiguration-config/pom.xml [hunk 3]
```diff
@@ -26,4 +26,79 @@
...
+                  ${project.basedir}/javadocTemp/README.md
+                </echo>
+                <copy file="${project.basedir}/../README.md" tofile="${project.basedir}/javadocTemp/README.md" failonerror="false"/>
```
Comment (chenrujun) [inline sdk/appconfiguration/azure-spring-cloud-starter-appconfiguration-config/pom.xml:88]: `${project.basedir}/../README.md` -> `${project.basedir}/README.md`.

File: sdk/appconfiguration/azure-spring-cloud-starter-appconfiguration-config/pom.xml [hunk 4]
```diff
@@ -26,4 +26,79 @@
...
+                <echo>Copying ${project.basedir}/../README.md to
```
Comment (chenrujun) [inline sdk/appconfiguration/azure-spring-cloud-starter-appconfiguration-config/pom.xml:89]: `${project.basedir}/../README.md` -> `${project.basedir}/README.md`.

File: sdk/appconfiguration/azure-spring-cloud-starter-appconfiguration-config/pom.xml [hunk 5]
```diff
@@ -26,4 +26,79 @@
...
+                  ${project.basedir}/sourceTemp/README.md
+                </echo>
+                <copy file="${project.basedir}/../README.md" tofile="${project.basedir}/sourceTemp/README.md" failonerror="false"/>
```
Comment (chenrujun) [inline sdk/appconfiguration/azure-spring-cloud-starter-appconfiguration-config/pom.xml:92]: `${project.basedir}/../README.md` -> `${project.basedir}/README.md`.

**Review by chenrujun — COMMENTED** [review]
The path of `README.md` is not right.

**Pull request merged**